<h1>Activity 2: UK ROAD ACCIDENT</h1>
<hr/>
<h3>Analyst: Edwin Viñas</h3>

<h1>Import Libraries</h1>

In [65]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import warnings
warnings.filterwarnings('ignore')

<h1>Convert the DataSet into DataFrame</h1>

In [66]:
accident = pd.read_csv("datasets//uk_road_accident.csv")

In [67]:
accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

In [68]:
accident.columns

Index(['Index', 'Accident_Severity', 'Accident Date', 'Latitude',
       'Light_Conditions', 'District Area', 'Longitude',
       'Number_of_Casualties', 'Number_of_Vehicles', 'Road_Surface_Conditions',
       'Road_Type', 'Urban_or_Rural_Area', 'Weather_Conditions',
       'Vehicle_Type'],
      dtype='object')

<h1>Checking for Null Values</h1>

In [69]:
accident.isnull()

Index  Accident_Severity  Accident Date  Latitude  Light_Conditions  \
0       False              False          False     False             False   
1       False              False          False     False             False   
2       False              False          False     False             False   
3       False              False          False     False             False   
4       False              False          False     False             False   
...       ...                ...            ...       ...               ...   
660674  False              False          False     False             False   
660675  False              False          False     False             False   
660676  False              False          False     False             False   
660677  False              False          False     False             False   
660678  False              False          False     False             False   

        District Area  Longitude  Number_of_Casualties  Number_of_Vehicles  \
0               False      False                 False               False   
1               False      False                 False               False   
2               False      False                 False               False   
3               False      False                 False               False   
4               False      False                 False               False   
...               ...        ...                   ...                 ...   
660674          False      False                 False               False   
660675          False      False                 False               False   
660676          False      False                 False               False   
660677          False      False                 False               False   
660678          False      False                 False               False   

        Road_Surface_Conditions  Road_Type  Urban_or_Rural_Area  \
0                         False      False                False   
1                         False      False                False   
2                         False       True                False   
3                         False      False                False   
4                         False       True                False   
...                         ...        ...                  ...   
660674                    False      False                False   
660675                    False      False                False   
660676                    False      False                False   
660677                    False      False                False   
660678                    False      False                False   

        Weather_Conditions  Vehicle_Type  
0                    False         False  
1                    False         False  
2                     True         False  
3                    False         False  
4                     True         False  
...                    ...           ...  
660674               False         False  
660675               False         False  
660676               False         False  
660677               False         False  
660678               False         False  

[660679 rows x 14 columns]

<h1>Descriptive Analytics</h1>

In [70]:
accident.describe()

Latitude      Longitude  Number_of_Casualties  Number_of_Vehicles
count  660654.000000  660653.000000         660679.000000       660679.000000
mean       52.553866      -1.431210              1.357040            1.831255
std         1.406922       1.383330              0.824847            0.715269
min        49.914430      -7.516225              1.000000            1.000000
25%        51.490690      -2.332291              1.000000            1.000000
50%        52.315641      -1.411667              1.000000            2.000000
75%        53.453452      -0.232869              1.000000            2.000000
max        60.757544       1.762010             68.000000           32.000000

<h1>Check DataFrame Info</h1>

In [71]:
accident.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660654 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660653 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  659953 non-null  object 
 10  Road_Type                656159 non-null  object 
 11  Urban_or_Rural_Area      660664 non-null  object 
 12  Weather_Conditions       646551 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

<h1>Filling Up The Null Values</h1>

In [72]:
accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [73]:
accident['Latitude']

0         51.506187
1         51.495029
2         51.517715
3         51.495478
4         51.488576
            ...    
660674    57.374005
660675    57.232273
660676    57.585044
660677    57.214898
660678    57.575210
Name: Latitude, Length: 660679, dtype: float64

In [74]:
accident['Latitude'].mean()

np.float64(52.553865761110956)

In [75]:
accident['Latitude'] = accident['Latitude'].fillna(accident['Latitude'].mean())

In [76]:
accident['Longitude']

0        -0.209082
1        -0.173647
2        -0.210215
3        -0.202731
4        -0.192487
            ...   
660674   -3.467828
660675   -3.809281
660676   -3.862727
660677   -3.823997
660678   -3.895673
Name: Longitude, Length: 660679, dtype: float64

In [77]:
accident['Longitude'].mean()

np.float64(-1.4312103685020727)

In [78]:
accident['Longitude'] = accident['Longitude'].fillna(accident['Longitude'].mean())

In [79]:
accident['Road_Surface_Conditions']

0                  Dry
1          Wet or damp
2                  Dry
3                  Dry
4                  Dry
              ...     
660674             Dry
660675    Frost or ice
660676    Frost or ice
660677     Wet or damp
660678     Wet or damp
Name: Road_Surface_Conditions, Length: 660679, dtype: object

In [80]:
accident['Road_Surface_Conditions'].mode()

0    Dry
Name: Road_Surface_Conditions, dtype: object

In [81]:
accident['Road_Surface_Conditions'] = accident['Road_Surface_Conditions'].fillna(accident['Road_Surface_Conditions'].mode()[0])

In [82]:
accident['Road_Type']

0         Single carriageway
1         Single carriageway
2                        NaN
3         Single carriageway
4                        NaN
                 ...        
660674    Single carriageway
660675    Single carriageway
660676    Single carriageway
660677    Single carriageway
660678      Dual carriageway
Name: Road_Type, Length: 660679, dtype: object

In [83]:
accident['Road_Type'].mode()

0    Single carriageway
Name: Road_Type, dtype: object

In [84]:
accident['Road_Type'] = accident['Road_Type'].fillna(accident['Road_Type'].mode()[0])

In [85]:
accident['Urban_or_Rural_Area']

0         Urban
1         Urban
2         Urban
3         Urban
4         Urban
          ...  
660674    Rural
660675    Rural
660676    Rural
660677    Rural
660678    Rural
Name: Urban_or_Rural_Area, Length: 660679, dtype: object

In [86]:
accident['Urban_or_Rural_Area'].mode()

0    Urban
Name: Urban_or_Rural_Area, dtype: object

In [87]:
accident['Urban_or_Rural_Area'] = accident['Urban_or_Rural_Area'].fillna(accident['Urban_or_Rural_Area'].mode()[0])

In [88]:
accident['Weather_Conditions']

0            Fine no high winds
1         Raining no high winds
2                           NaN
3            Fine no high winds
4                           NaN
                  ...          
660674       Fine no high winds
660675       Fine no high winds
660676       Fine no high winds
660677       Fine no high winds
660678    Snowing no high winds
Name: Weather_Conditions, Length: 660679, dtype: object

In [89]:
accident['Weather_Conditions'].mode()

0    Fine no high winds
Name: Weather_Conditions, dtype: object

In [90]:
accident['Weather_Conditions'] = accident['Weather_Conditions'].fillna(accident['Weather_Conditions'].mode()[0])

In [91]:
accident.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
dtype: int64

<h1>Changing Dtypes</h1>

In [92]:
accident.dtypes

Index                       object
Accident_Severity           object
Accident Date               object
Latitude                   float64
Light_Conditions            object
District Area               object
Longitude                  float64
Number_of_Casualties         int64
Number_of_Vehicles           int64
Road_Surface_Conditions     object
Road_Type                   object
Urban_or_Rural_Area         object
Weather_Conditions          object
Vehicle_Type                object
dtype: object

In [93]:
accident['Index'] = accident['Index'].astype("category")
accident['Accident_Severity'] = accident['Accident_Severity'].astype("category")
accident['Accident Date'] = accident['Accident Date'].astype("category")
accident['Light_Conditions'] = accident['Light_Conditions'].astype("category")
accident['District Area'] = accident['District Area'].astype("category")
accident['Road_Surface_Conditions'] = accident['Road_Surface_Conditions'].astype("category")
accident['Road_Type'] = accident['Road_Type'].astype("category")
accident['Urban_or_Rural_Area'] = accident['Urban_or_Rural_Area'].astype("category")
accident['Weather_Conditions'] = accident['Weather_Conditions'].astype("category")
accident['Vehicle_Type'] = accident['Vehicle_Type'].astype("category")

In [94]:
accident.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

<h3>Clearing any Inconsistencies with the Data Set</h3>

In [95]:
accident['Accident Date'] = accident['Accident Date'].str.strip()
accident['Accident Date'] = accident['Accident Date'].astype('str')
accident['Accident Date'] = accident['Accident Date'].str.replace('/','-')

In [96]:
accident['Accident Date'] = pd.to_datetime(accident['Accident Date'],dayfirst = True, errors = 'coerce')

<h3>Etracting Date Information using pandas Date Time</h3>

In [97]:

accident["Accident Date"] = pd.to_datetime(accident["Accident Date"], errors="coerce")


accident['Year'] = accident['Accident Date'].dt.year
accident['Month'] = accident['Accident Date'].dt.month
accident['Day'] = accident['Accident Date'].dt.day
accident['DayOfWeek'] = accident['Accident Date'].dt.dayofweek


In [98]:
accident.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
Year                       0
Month                      0
Day                        0
DayOfWeek                  0
dtype: int64

## 1. What is the distribution of accident severity?

In [99]:
accident['Accident_Severity'].value_counts()

Accident_Severity
Slight     563801
Serious     88217
Fatal        8661
Name: count, dtype: int64

<h3>Insight: Most accidents are classified as Slight, while Serious and Fatal accidents are less frequent, showing that severe crashes are relatively rare.</h3>

### 2. What is the average number of casualties per accident severity?

In [100]:
accident.groupby('Accident_Severity')['Number_of_Casualties'].mean()

Accident_Severity
Fatal      1.903129
Serious    1.467280
Slight     1.331402
Name: Number_of_Casualties, dtype: float64

<h3>Insight: Fatal accidents, although less common, tend to result in more casualties per incident.</h3>

<h3>3. Do more accidents occur in urban or rural areas?</h3>

In [101]:
accident['Urban_or_Rural_Area'].value_counts(dropna=False)

Urban_or_Rural_Area
Urban          421678
Rural          238990
Unallocated        11
Name: count, dtype: int64

<h3>Insight: Urban areas usually record more accidents due to higher traffic density; Rural areas may have fewer crashes but at higher speeds, often more severe.</h3>

<h3>4. Which road types report the highest number of accidents?</h3>

In [102]:
accident['Road_Type'].value_counts(dropna=False).head(10)

Road_Type
Single carriageway    496663
Dual carriageway       99424
Roundabout             43992
One way street         13559
Slip road               7041
Name: count, dtype: int64

<h3>Insight: Single carriageways often dominate counts, possibly because they carry mixed traffic without physical separation.</h3>

<h3>5. Which road surface condition has the highest accident rate?</h3>

In [103]:
accident['Road_Surface_Conditions'].value_counts()

Road_Surface_Conditions
Dry                     448547
Wet or damp             186708
Frost or ice             18517
Snow                      5890
Flood over 3cm. deep      1017
Name: count, dtype: int64

<h3>Insight: Dry roads show the highest accident count, likely because most travel occurs in dry conditions.</h3>

<h3>6. Do most accidents occur in daylight or darkness?</h3>

In [104]:
accident['Light_Conditions'].value_counts()

Light_Conditions
Daylight                       484880
Darkness - lights lit          129335
Darkness - no lighting          37437
Darkness - lighting unknown      6484
Darkness - lights unlit          2543
Name: count, dtype: int64

<H3>Insight: Accidents are more frequent in daylight, but night crashes are often more severe.</H3>

<h3>7. Which light condition has the highest fatality rate?</h3>

In [105]:
accident[accident['Accident_Severity']=='Fatal']['Light_Conditions'].value_counts()

Light_Conditions
Daylight                       5076
Darkness - lights lit          1860
Darkness - no lighting         1612
Darkness - lighting unknown      68
Darkness - lights unlit          45
Name: count, dtype: int64

<h3>Insight: Darkness without streetlights shows a higher share of fatal accidents, highlighting visibility issues</h3>

<h3>8. Which weather conditions are most common in accidents?</h3>

In [106]:
accident['Weather_Conditions'].value_counts().head(10)

Weather_Conditions
Fine no high winds       535013
Raining no high winds     79696
Other                     17150
Raining + high winds       9615
Fine + high winds          8554
Snowing no high winds      6238
Fog or mist                3528
Snowing + high winds        885
Name: count, dtype: int64

<h3>Insight: Fine weather accounts for the majority of accidents due to higher traffic, though bad weather raises accident severity.</h3>

<h3>9. How does weather condition affect average casualties?</h3>

In [107]:
accident.groupby('Weather_Conditions')['Number_of_Casualties'].mean()

Weather_Conditions
Fine + high winds        1.386018
Fine no high winds       1.347397
Fog or mist              1.452948
Other                    1.354869
Raining + high winds     1.416641
Raining no high winds    1.408214
Snowing + high winds     1.418079
Snowing no high winds    1.341776
Name: Number_of_Casualties, dtype: float64

<h3>Insight: Fog or mist tends to have the highest average casualties per accident because of reduced visibility.</h3>

<h3>10. Which vehicle types are most often involved in accidents?</h3>

In [108]:
accident['Vehicle_Type'].value_counts().head(10)

Vehicle_Type
Car                                      497992
Van / Goods 3.5 tonnes mgw or under       34160
Bus or coach (17 or more pass seats)      25878
Motorcycle over 500cc                     25657
Goods 7.5 tonnes mgw and over             17307
Motorcycle 125cc and under                15269
Taxi/Private hire car                     13294
Motorcycle over 125cc and up to 500cc      7656
Motorcycle 50cc and under                  7603
Goods over 3.5t. and under 7.5t            6096
Name: count, dtype: int64

<h3>Insight: Cars dominate accident records, but motorcycles and bicycles are more vulnerable in serious crashes.</h3>

<h3>11. Which columns have the most missing data?</h3>

In [109]:
accident.isnull().sum().sort_values(ascending=False)

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
Year                       0
Month                      0
Day                        0
DayOfWeek                  0
dtype: int64

<h3>Insight: Weather and road type fields have noticeable missing values, which can affect reliability of analysis.</h3>

<h3>12. Do fatal accidents occur more often in urban or rural areas?</h3>

In [110]:
accident[accident['Accident_Severity']=='Fatal']['Urban_or_Rural_Area'].value_counts()

Urban_or_Rural_Area
Rural          5601
Urban          3060
Unallocated       0
Name: count, dtype: int64

<h3>Insight:Rural areas, though less frequent overall, often see more fatal outcomes due to high-speed crashes.</h3>

<h3>13. Which vehicle types are most common in fatal accidents?</h3>

In [111]:
accident[accident['Accident_Severity']=='Fatal']['Vehicle_Type'].value_counts().head(10)

Vehicle_Type
Car                                      6577
Van / Goods 3.5 tonnes mgw or under       467
Motorcycle over 500cc                     339
Bus or coach (17 or more pass seats)      325
Goods 7.5 tonnes mgw and over             216
Motorcycle 125cc and under                189
Taxi/Private hire car                     155
Motorcycle over 125cc and up to 500cc     105
Motorcycle 50cc and under                  95
Other vehicle                              70
Name: count, dtype: int64

<h3>Insight: Cars are most frequent, but motorcycles show disproportionately high fatality risks.</h3>

<h3>14.  Which weather conditions are most common in fatal accidents?</h3>

In [112]:
accident[accident['Accident_Severity']=='Fatal']['Weather_Conditions'].value_counts().head(10)

Weather_Conditions
Fine no high winds       7207
Raining no high winds     848
Fine + high winds         175
Other                     165
Raining + high winds      145
Fog or mist                82
Snowing no high winds      36
Snowing + high winds        3
Name: count, dtype: int64

<h3>Insight: Fatal crashes are more frequent in clear weather, showing traffic exposure plays a large role.</h3>

<h3>15. Which district areas report the most accidents?</h3>

In [113]:
accident['District Area'].value_counts().head(10)

District Area
Birmingham          13491
Leeds                8898
Manchester           6720
Bradford             6212
Sheffield            5710
Westminster          5706
Liverpool            5587
Glasgow City         4942
Bristol, City of     4819
Kirklees             4690
Name: count, dtype: int64

<h3>Insight: Highly populated urban districts are hotspots for accidents.</h3>

<h3>16. Which months record the highest number of accidents?</h3>

In [114]:
accident['Accident Date'] = pd.to_datetime(accident['Accident Date'], errors='coerce', dayfirst=True)
accident['Month'] = accident['Accident Date'].dt.month
accident['Month'].value_counts().sort_index()

Month
1     52872
2     49491
3     54086
4     51744
5     56352
6     56481
7     57445
8     53913
9     56455
10    59580
11    60424
12    51836
Name: count, dtype: int64

<h3>Insight: Seasonal peaks in accidents may occur during summer holidays or winter weather conditions.</h3>

<h3>17. What is the distribution of casualties per accident?:</h3>

In [115]:
accident['Number_of_Casualties'].value_counts().sort_index().head(15)

Number_of_Casualties
1     504041
2     107971
3      30776
4      11156
5       4123
6       1525
7        539
8        214
9        112
10        58
11        36
12        21
13        24
14        13
15        11
Name: count, dtype: int64

<h3>Insight: The majority of accidents involve a single casualty, but some accidents cause multiple injuries.</h3>

<h3>18. What is the distribution of the number of vehicles involved in accidents?</h3>

In [116]:
accident['Number_of_Vehicles'].value_counts().sort_index()

Number_of_Vehicles
1     200787
2     391995
3      52806
4      11300
5       2464
6        758
7        296
8        147
9         57
10        27
11        12
12         7
13         6
14         9
15         1
16         4
19         1
28         1
32         1
Name: count, dtype: int64

<h3>Isight: Most accidents involve 2 vehicles, but multi-vehicle collisions lead to higher casualty counts.</h3>

<h3>19. Do buses or cars result in more casualties per accident?</h3>

In [117]:
accident.groupby('Vehicle_Type')['Number_of_Casualties'].mean().sort_values(ascending=False).head(10)

Vehicle_Type
Data missing or out of range            1.500000
Pedal cycle                             1.370558
Taxi/Private hire car                   1.368663
Goods over 3.5t. and under 7.5t         1.362861
Motorcycle over 500cc                   1.359434
Car                                     1.358841
Van / Goods 3.5 tonnes mgw or under     1.354537
Goods 7.5 tonnes mgw and over           1.351881
Bus or coach (17 or more pass seats)    1.349216
Minibus (8 - 16 passenger seats)        1.345648
Name: Number_of_Casualties, dtype: float64

<h3>Insight: Buses tend to involve more casualties per accident because they carry multiple passengers.</h3>

<h3>20.  Which days of the week see the most accidents?</h3>

In [118]:
accident['DayOfWeek'] = accident['Accident Date'].dt.day_name()
accident['DayOfWeek'].value_counts()

DayOfWeek
Saturday     107178
Wednesday     99558
Thursday      99511
Friday        97900
Tuesday       94550
Sunday        89302
Monday        72680
Name: count, dtype: int64

<h3>Insight: Weekdays often show more accidents due to commuter traffic, while weekends may reflect leisure travel risks.</h3>

<H3>21. Accident severity based on area of loacation(Rural or Urban)

In [119]:
severity = accident.groupby(['Accident_Severity','Urban_or_Rural_Area']).size()

In [120]:
severity.unstack()

Urban_or_Rural_Area   Rural  Unallocated   Urban
Accident_Severity                               
Fatal                  5601            0    3060
Serious               37312            1   50904
Slight               196077           10  367714


<h3>Insight: There are more fatal accidents in rural areas, serious and slight accidents on Urban areas</h3>

<H3>22. Accident severity based on vehicle type, what type of vehicle is likely to have a fatal accident?</H3>

In [121]:
severity2 = accident.groupby(['Accident_Severity','Vehicle_Type']).size()

In [122]:
severity2.unstack()

Vehicle_Type       Agricultural vehicle  Bus or coach (17 or more pass seats)  \
Accident_Severity                                                               
Fatal                                21                                   325   
Serious                             282                                  3373   
Slight                             1644                                 22180   

Vehicle_Type          Car  Data missing or out of range  \
Accident_Severity                                         
Fatal                6577                             0   
Serious             66461                             0   
Slight             424954                             6   

Vehicle_Type       Goods 7.5 tonnes mgw and over  \
Accident_Severity                                  
Fatal                                        216   
Serious                                     2321   
Slight                                     14770   

Vehicle_Type       Goods over 3.5t. and under 7.5t  \
Accident_Severity                                    
Fatal                                           67   
Serious                                        857   
Slight                                        5172   

Vehicle_Type       Minibus (8 - 16 passenger seats)  \
Accident_Severity                                     
Fatal                                            29   
Serious                                         276   
Slight                                         1671   

Vehicle_Type       Motorcycle 125cc and under  Motorcycle 50cc and under  \
Accident_Severity                                                          
Fatal                                     189                         95   
Serious                                  2031                       1014   
Slight                                  13049                       6494   

Vehicle_Type       Motorcycle over 125cc and up to 500cc  \
Accident_Severity                                          
Fatal                                                105   
Serious                                             1014   
Slight                                              6537   

Vehicle_Type       Motorcycle over 500cc  Other vehicle  Pedal cycle  \
Accident_Severity                                                      
Fatal                                339             70            6   
Serious                             3457            767           39   
Slight                             21861           4800          152   

Vehicle_Type       Ridden horse  Taxi/Private hire car  \
Accident_Severity                                        
Fatal                         0                    155   
Serious                       0                   1771   
Slight                        4                  11368   

Vehicle_Type       Van / Goods 3.5 tonnes mgw or under  
Accident_Severity                                       
Fatal                                              467  
Serious                                           4554  
Slight                                           29139

<h3>Insight: There is a high chance that cars can have fatal accidents, followed by motorcyles and vans</h3>


<H3>23. Number of accident severity</H3>

In [123]:
accident['Accident_Severity'].value_counts()

Accident_Severity
Slight     563801
Serious     88217
Fatal        8661
Name: count, dtype: int64



<h3>Insight: Accident severity has the following values: Slight 563,801 Serious 88,217 Fatal 8,661 </h3>

<H3>24. Most common roadtype accident?</H3>

In [124]:
accident['Road_Type'].value_counts()

Road_Type
Single carriageway    496663
Dual carriageway       99424
Roundabout             43992
One way street         13559
Slip road               7041
Name: count, dtype: int64

<h3>Insight: There have been 75.17% of accidents recorded in Single carriageway type of road</h3>

<H3>25 Accident by weather condition?</H3>

In [125]:
accident['Weather_Conditions'].value_counts()

Weather_Conditions
Fine no high winds       535013
Raining no high winds     79696
Other                     17150
Raining + high winds       9615
Fine + high winds          8554
Snowing no high winds      6238
Fog or mist                3528
Snowing + high winds        885
Name: count, dtype: int64

<h3>Insight: There have been 80.56% of accidents recorded in Fine no high winds type of weather condition</h3>

<h1>Data Visualization</h1>

In [146]:
kingston_df = accident[accident['District Area'] == 'Kingston upon Thames']

In [153]:
import folium 
from folium.plugins import HeatMap

locations = list(zip(kingston_df['Latitude'], kingston_df['Longitude']))
Kingston_Map = folium.Map(location = [kingston_df['Latitude'].mean(), kingston_df['Longitude'].mean()],zoom_start =10)
HeatMap(locations).add_to(Kingston_Map)

In [154]:
Kingston_Map

<h1>DATA VISUALIZATION USING SEABORN</h1>

In [151]:
import seaborn as sns

In [156]:
sns.barplot(data=accident, y='Number_of_Casualities', x = 'Month')
plt.gcf().set_size_inches(10,6)
plt.show()


ValueError: Could not interpret value `Number_of_Casualities` for `y`. An entry with this name does not appear in `data`.